test/integration-new/snat/snat_suite_test.go

```diff
@@ -0,0 +1,146 @@
+	testUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	f                                                     *framework.Framework
+	props                                                 utils.NodeGroupProperties
+	primaryNodeInPublicSubnet, primaryNodeInPrivateSubnet v1.Node
+	privateSubnetId                                       string
+)
+
+func TestSnat(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Snat Suite")
+}
+
+var _ = BeforeSuite(func() {
+	f = framework.New(framework.GlobalOptions)
+	By("creating test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		CreateNamespace(testUtils.DefaultTestNamespace)
+
+	By("Getting existing nodes in the cluster")
+	nodes, err := f.K8sResourceManagers.NodeManager().GetAllNodes()
+	Expect(err).ToNot(HaveOccurred())
+
+	By("verifying more than 1 nodes are present for the test")
+	Expect(len(nodes.Items)).Should(BeNumerically(">", 1))
+
+	// Set the primary node for testing
+	primaryNodeInPublicSubnet = nodes.Items[0]
+
+	By("Getting Public and Private subnets")
+	vpcConfig, err := utils.GetClusterVPCConfig(f)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(len(vpcConfig.PublicSubnetList)).To(BeNumerically(">", 0))
+	Expect(len(vpcConfig.PrivateSubnetList)).To(BeNumerically(">", 0))
+
+	msg := fmt.Sprintf("Creating a keyPair with name: %s if it doesn't exist", f.Options.KeyPair)
+	By(msg)
+	keyPairOutput, _ := f.CloudServices.EC2().DescribeKey(f.Options.KeyPair)
+
+	exists := false
+	if keyPairOutput != nil {
+		for _, keyPair := range keyPairOutput.KeyPairs {
+			if *keyPair.KeyName == f.Options.KeyPair {
+				exists = true
+				break
+			}
+		}
+	}
+
+	if exists {
+		fmt.Println("KeyPair already exists")
+	} else {
+		fmt.Println("KeyPair doesn't exist, will be created")
+		_, err := f.CloudServices.EC2().CreateKey(f.Options.KeyPair)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	privateSubnetId = vpcConfig.PrivateSubnetList[0]
+
+	By("Getting Cluster Security Group Id")
+	out, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+	Expect(err).NotTo(HaveOccurred())
+
+	clusterSecurityGroupId := out.Cluster.ResourcesVpcConfig.ClusterSecurityGroupId
+
+	msg = fmt.Sprintf("Deploying a self managed nodegroup of size 1 in private subnet %s", privateSubnetId)
+	By(msg)
+	props = utils.NodeGroupProperties{
+		NgLabelKey:    "test-label-key",
+		NgLabelVal:    "test-label-val",
+		AsgSize:       1,
+		NodeGroupName: "snat-test-ng",
+		Subnet: []string{
+			privateSubnetId,
+		},
+		InstanceType: "m5.large",
+		KeyPairName:  f.Options.KeyPair,
+	}
+
+	err = utils.CreateAndWaitTillSelfManagedNGReady(f, props)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Fetching existing Security Groups from the newly created node group instance")
+	instanceOutput, err := f.CloudServices.EC2().DescribeInstancesWithFilters(map[*string][]*string{
```

Comment:
This will return the new instance created by the test + any nodes present in the private subnet as a list in any order. Later in the test we are picking 0th index of the list output which could be the existing instance as well right?

Correct way to me looks like - Get the newly created node and get instance ID from it and use that.